### DISCUSSION TITLE
Replace caps lock with escape with just i3?

### DISCUSSION THREAD

**Discussion by ghost** [i3/i3#4570] [Q&A] [answered]
Is it possible to replace caps lock with escape using only i3 config file?

**Comment by Airblader** [comment, marked as answer]
You can use [Xmodmap](https://wiki.archlinux.org/title/xmodmap) for this. And yes you could trigger a command from the i3 config, but using an Xmodmap config would probably be more sensible.

**Reply by ghost** [reply]
I used better method. What I did was edit /usr/share/X11/xkb/symbols/pc file. I changed this line:
`key <CAPS> {    [ Escape        ]   };`
Works pefectly!

**Reply by ghost** [reply]
But i didnt use i3 for that.

**Reply by dojero** [reply]
I'm not sure yours is a 'better method', though it is certainly a way to go. Xmodmap can be used to reassign keys for specific users, thus leaving the computer without a reassignment when another user is on the system. Your altering of the /usr/share file will mean that any user will have the key re-assigned. While that may be better in your use case, it's not necessarily better for everyone.

**Reply by orestisfl** [reply]
Additionally, if this file is managed by a package manager, the change could be reverted at some point.

**Comment by ghost** [comment]
fwiw this is the only solution that worked on my system, in i3 add: exec
setxkbmap -option caps:swapescape

On Sat, Oct 9, 2021 at 11:58 AM Orestis Floros ***@***.***>
wrote:

> Additionally, if this file is managed by a package manager, the change
> could be reverted at some point.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/i3/i3/discussions/4570#discussioncomment-1452932>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMYM3JKYANZIJXW7O2HZHW3UGBRC5ANCNFSM5FUNNCTA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 

All the world's problems can be fixed in a garden - Geoff Lawton

**Reply by alexbenisch** [reply]
I have a similar question regarding caps, as I switched recently there might be a better way I haven't found yet. 
On xfce I disabled caps and used only the keycode 66. With this I am almost where I want be

```
exec xmodmap - e 'clear lock' 
bindcode 66 exec copy toggle
```
But it is not really persistent and
`
setxkbmap -option caps:none
`
I can manually disable caps again. But what I want is making it persistent always everywhere.
How can I achieve this?
Thanks

Edit:
I changed it by placing
`clear lock
`in
`.Xmodmap 
`
and
`exec xmodmap ~/.Xmodmap `
in
`~/.config/i3/config
`

**Reply by dojero** [reply]
You can make your xmodmap changes permanent by putting them into an ~/.Xmodmap file. To assure that the file is used by xorg, you will have to add some lines to your .xinitrc file. All of this is well explained in a variety of places. See, for example, the Arch Linux Wiki page on xmodmap ([](url)https://wiki.archlinux.org/title/Xmodmap).
